### ISSUE TITLE
[Question] - Filter on Relation fields

### ISSUE THREAD

**Issue by rshriyan** [doug-martin/nestjs-query#223]
Hello Doug

I started working on GraphQL just this week so I am a total beginner. I came across your abstraction for CRUD operation's and it seems to fit my requirement very well. I do have a question for you. I have the following relation(Psuedo code):

```
class Car {
  name: string;
  @OnetoOne() manufacturer: Manufacturer
}

class Manufacturer {
  name: string;
}
```

Frontend client is showing all of this data as flat list meaning:

`Car Name | Manufacturer Name`

With the standard setup and by using `Relations` from your library, everything works well. However those fields should be filterable specifically the manufacturer name. Does this require me to implement my own CRUD resolver and sort of have a flat list DTO to represent the flat list the frontend expects ?

**Comment by doug-martin** [comment]
This package doesnt currently support filtering on relation values.

One way to achieve this is to think about the problem slightly differently, instead of using car as root of your query instead query for manufacturers with the name you want to query for and include their cars.

**Comment by renehauck** [comment]
I guess the problem is that in my opinion it is not always possible to solve the problem that way. :disappointed: 
For example if you want to implement a table in a frontend where one entry has many relations (A person who has a contact information, an address etc). 
If you now want to filter/sort by address, would you already have a problem, or do I get something wrong?
For me and I think others, this would be a very important feature. :grinning:

**Comment by bestickley** [comment]
@doug-martin, would it be possible to update the [FilterBuilder ](https://github.com/doug-martin/nestjs-query/blob/master/packages/core/src/helpers/filter.builder.ts) to allow for something like this:
`{ user: { role { is: "Admin" } }`?

**Comment by doug-martin** [comment]
@rshriyan @bestickley  this enhancement has been published under `v0.16.0`. 

You can read more about this feature in the following docs

* https://doug-martin.github.io/nestjs-query/docs/graphql/relations
* https://doug-martin.github.io/nestjs-query/docs/migration-guides/v0.15.x-to-v0.16.x

Thank you for the enhancement request!
